sdk/go/common/resource/plugin/provider_plugin.go

```diff
@@ -635,32 +676,33 @@ func (p *provider) Check(urn resource.URN,
 	label := fmt.Sprintf("%s.Check(%s)", p.label(), urn)
 	logging.V(7).Infof("%s executing (#olds=%d,#news=%d", label, len(olds), len(news))
 
-	// Get the RPC client and ensure it's configured.
-	client, err := p.getClient()
+	// Ensure that the plugin is configured.
+	client := p.clientRaw
+	pcfg, err := p.awaitConfig(context.Background())
```

Comment:
sigh these provider methods really ought to take a context, something for later it'll be an effort to fix up